xline/src/server/lease_server.rs

```diff
@@ -286,47 +255,33 @@ where
     ) -> Result<tonic::Response<Self::LeaseKeepAliveStream>, tonic::Status> {
         debug!("Receive LeaseKeepAliveRequest {:?}", request);
         let request_stream = request.into_inner();
-        let stream = if self.is_leader() {
-            self.leader_keep_alive(request_stream).await
-        } else {
-            self.follower_keep_alive(request_stream).await?
+        let stream = loop {
+            if self.lease_storage.is_primary() {
+                break self.leader_keep_alive(request_stream).await;
+            }
+            let leader_id = self.client.get_leader_id().await;
+            // double check
```

Comment:
why do we need double check here?

Reply:
Let's assume that the current server is a candidate and the lease store it has is not a primary lessor. So the self.client will wait for the new leader id to return. If the candidate lost the election, it should redirect this request to the real leader, otherwise, it should deal with this request by itself.